src/os/bluestore/BlueStore.cc

```diff
@@ -4737,7 +4737,7 @@ int BlueStore::_do_read(
       for (auto reg : b2r_it->second) {
 	// determine how much of the blob to read
 	uint64_t chunk_size = bptr->get_blob().get_chunk_size(
-	  csum_type != bluestore_blob_t::CSUM_NONE, block_size);
+	  csum_type.load() != bluestore_blob_t::CSUM_NONE, block_size);
```

Comment:
I think this doesn't make much sense.  bluestore_csum_type is a kind of csum.. and on read we do'nt get to choose.  I think we need to either just verify csum if it exists or introduce a new option bluestore_csum_verify_on_read that can be disabled if the admin really wants to ignore it.